File: 백준/Silver/2002. 추월/추월.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        Queue<String> inCars = new LinkedList<>();
        Queue<String> outCars = new LinkedList<>();

        for (int i = 0; i < N; i++) {
            inCars.offer(br.readLine());
        }

        for (int i = 0; i < N; i++) {
            outCars.offer(br.readLine());
        }

        int count = 0;

        while (!outCars.isEmpty()) {
            String outCar = outCars.poll();

            if (inCars.peek().equals(outCar)) {
                inCars.poll(); 
            } else {
                count++;
                removeCar(inCars, outCar);
            }
        }

        System.out.println(count);
    }
    
    private static void removeCar(Queue<String> queue, String car) {
        Queue<String> temp = new LinkedList<>();
        while (!queue.isEmpty()) {
            String currentCar = queue.poll();
            if (!currentCar.equals(car)) {
                temp.offer(currentCar);
            }
        }
        queue.addAll(temp);
    }
}

File: 백준/Silver/2002. 추월/추월.py
import sys
from collections import deque

def main():
    N = int(sys.stdin.readline())
    in_cars = deque()
    out_cars = deque()

    for _ in range(N):
        in_cars.append(sys.stdin.readline())

    for _ in range(N):
        out_cars.append(sys.stdin.readline())

    count = 0

    while out_cars:
        out_car = out_cars.popleft()

        if in_cars[0] == out_car:
            in_cars.popleft()
        else:
            count += 1
            remove_car(in_cars, out_car)

    print(count)

def remove_car(queue, car):
    temp = deque()
    while queue:
        current_car = queue.popleft()
        if current_car != car:
            temp.append(current_car)
    queue.extend(temp)

if __name__ == "__main__":
    main()